core/src/runtime/hooks.rs

```diff
@@ -0,0 +1,100 @@
+use core::convert::TryInto;
+use std::collections::HashMap;
+
+use k256::ecdsa::{RecoveryId, Signature, VerifyingKey};
+use k256::elliptic_curve::ops::Invert;
+
+use super::Runtime;
+
+type HookId = u32;
+type BoxedHook<'a> = Box<dyn Fn(HookEnv, &[u8]) -> Vec<Vec<u8>> + 'a>;
+
+// Ensure this value is synced with the values in `zkvm/precompiles/src/io.rs`
```

Comment:
we might be able to use `static_assertions` crate to ensure they are the same

Reply:
I instead wrote a test in `hooks.rs` to check for equality.